packages/hooks/src/hooks/useStaticProvider.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { ethers } from 'ethers';
+import { StaticJsonRpcProvider } from '@ethersproject/providers';
+
+const createProvider = async (url: string) => {
+  const p = new ethers.providers.StaticJsonRpcProvider(url);
+  await p.ready;
+  return p;
+};
+
+/*
+ * @dev returns a static JSON RPC provider
+ * @param url - The url of the JSON RPC provider
+ * @credits scaffold-eth (https://github.com/scaffold-eth/scaffold-eth/blob/master/packages/react-app/src/hooks/useStaticJsonRPC.js)
+ */
+export function useStaticProvider(url: string | undefined) {
```

Comment:
why let url be undefined? It should probably be required right?
```suggestion
export function useStaticProvider(url: string) {
```

Reply:
So this is the issue with it: we are using this hook from `Provider.tsx` where it is optional to pass in a RPC URL which is then passed to this hook. I'm not sure how to call our hook conditionally

Reply:
Pushing a patch for now